frontend/src/packages/api/filesystem.ts

```diff
@@ -0,0 +1,92 @@
+import { CreateFilesystemEntryResponse, FilesystemEntry } from "./types/filesystem";
+import { APIError, EmptyAPIResponse } from "./types/general";
+
+// PostBody is an internal discriminated union of supported post bodies
+type PostBody = 
+    | { $type: "Record", body: Record<string, string> }
+    | { $type: "FormData", body: FormData }
```

Comment:
```suggestion
type PostBody = 
    | { $type: "Record"; body: Record<string, string> }
    | { $type: "FormData"; body: FormData };
```

while typescript lets you use either commas or semicolons to separate entries in a type, a quick search through the codebase seems like it *usually* uses semicolons, so it's good to be consistent